include/spdlog/details/async_log_helper.h

```diff
@@ -257,15 +256,11 @@ inline void spdlog::details::async_log_helper::flush(bool wait_for_q)
         wait_empty_q(); //return only make after the above flush message was processed
 }
 
-inline void spdlog::details::async_log_helper::worker_loop()
+inline bool spdlog::details::async_log_helper::worker_loop()
 {
     try
     {
```

Comment:
This is not a loop anymore. it is just a method wrapped into a function pointer that will be executed by the threads in the thread pool.